atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTable.java

```diff
@@ -118,6 +123,33 @@ public void create(byte[] tableMetadata) {
                 "INSERT INTO " + config.metadataTable().getQualifiedName() + " (table_name, table_size) VALUES (?, ?)");
     }
 
+    private void maybeAlterTableToHaveOverflow() {
+        try {
+            String shortTableName = oracleTableNameGetter.getInternalShortTableName(conns, tableRef);
+
+            if (config.alterTablesOrMetadataToMatch().contains(tableRef)
+                    && overflowTableHasMigrated()
+                    && overflowTableExists()
+                    && !overflowColumnExists(shortTableName)) {
+                alterTableToHaveOverflowColumn(shortTableName);
+            }
+        } catch (TableMappingNotFoundException e) {
+            log.warn("Table mapping expected but not found for table.", UnsafeArg.of("tableRef", tableRef), e);
+        }
+    }
+
+    private void alterTableToHaveOverflowColumn(String shortTableName) {
+        log.info(
+                "Altering table to have overflow column to match metadata.",
+                UnsafeArg.of("tableReference", tableRef),
+                UnsafeArg.of("shortTableName", shortTableName));
+        conns.get().executeUnregisteredQuery("ALTER TABLE " + shortTableName + " ADD (overflow NUMBER(38))");
```

Comment:
Perhaps worth a comment on the createTable that if you modify the overflow column schema then it should be modified here.